### PR TITLE
[DOC] CSD example matplotlib OO API

### LINKED ISSUE #10697: in CSD example, use matplotlib OO API instead of state machine
In `examples/time_frequency/compute_csd.py`, remove the calls to `plt.suptitle` and instead save the figure handle and use `fig.suptitle`.

### PR DESCRIPTION
**Pull request by sena-neuro**
#### Reference issue
Closes #10697. Currently, state machine API to matplotlib is used in [the CSD example](https://mne.tools/stable/auto_examples/time_frequency/compute_csd.html). In this PR, figure handles are saved and object-oriented API is used to plot CSD matrices.

#### Current CSD example
old_csd_ex

#### Proposed changes in CSD example after this PR
new_csd_ex

#### Are these changes ok?
-  I have saved each figure in different variables. (instead of just `fig` for all of them)
-  I have explained indexing after the call to `mne.time_frequency.CrossSpectralDensity.plot()` because I believe it was a bit awkward.

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
I personally find the new code more complicated for new comers

File: examples/time_frequency/compute_csd.py [hunk 1]
```diff
@@ -84,11 +82,16 @@
...
+fig_fft.suptitle('short-term Fourier transform')
+fig_fft.show()
+
+fig_mt = csd_mt.mean().plot(show=False)[0]
+fig_mt.suptitle('adaptive multitapers')
+fig_mt.show()
+
+fig_wav = csd_wav.mean().plot(show=False)[0]
+fig_wav.suptitle('Morlet wavelet transform')
+fig_wav.show()
```
Comment (drammock) [inline examples/time_frequency/compute_csd.py:97]:
```suggestion
plot_dict = {'Short-time Fourier transform': csd_fft,
             'Adaptive multitapers': csd_mt,
             'Morlet wavelet transform': csd_wav}

for title, csd in plot_dict.items():
    fig, = csd.mean().plot()
    fig.suptitle(title)
```

File: examples/time_frequency/compute_csd.py [hunk 2]
```diff
@@ -84,11 +82,16 @@
...
-csd_fft.mean().plot()
-plt.suptitle('short-term Fourier transform')
-
-csd_mt.mean().plot()
-plt.suptitle('adaptive multitapers')
-
-csd_wav.mean().plot()
-plt.suptitle('Morlet wavelet transform')
+# :func:`mne.time_frequency.CrossSpectralDensity.plot()` returns a list of
+# created figures. In this case, each returned list has one figure.
```
Comment (drammock) [inline examples/time_frequency/compute_csd.py:86]:
```suggestion
# created figures; in this case, each returned list has only one figure
# so we use a Python trick of including a comma after our variable name
# to assign the figure (not the list) to our ``fig`` variable:
```

**Review by drammock — COMMENTED** [review]
Here is how I would do it.  It involves less copy-paste code and avoids needing to `[0]`-index into the return value of `plot()`. Note also that I've corrected "short-term Fourier transform" to "short-time Fourier transform", so even if @agramfort doesn't like this version either, you should still correct that terminology in whichever version we end up with.

File: examples/time_frequency/compute_csd.py (hunk 2 above)
Comment (sena-neuro) [reply examples/time_frequency/compute_csd.py:86]: This is so much cleaner, thanks!

**Comment by drammock** [comment]
looks good:

https://output.circle-artifacts.com/output/job/f8ec3287-0ea3-4a93-a8d9-66cb0f69c811/artifacts/0/dev/auto_examples/time_frequency/compute_csd.html

Some of the failing CIs will be fixed by #11066; once that one is merged into `main` you should merge the `main` branch here @sena-neuro

**Comment by drammock** [comment]
thanks @sena-neuro!

**Pull request merged**